### PR TITLE
Docker (Local Cloud) + Testing

### PR DESCRIPTION
**Pull request by nicolas-kuechler**
Adds support to run on containers in a local Docker cluster (via the Docker API).

- Manages the Docker cluster via the Docker API
- Provides a standard image for Ubuntu 20.04 that is built locally and has support for custom Dockerfiles and (pre-built) images
- Fixes some non-determinism issues in the ETL of the test suite by adding an explicit sorting transformer in certain steps
- Adds docker-test Github Action workflow

### CODE REVIEW

**Comment by hiddely** [comment]
Question here: Do we want to set the `docker_image_id` by default in new projects? i.e., add to `exp_base` in the cookie-cutter template?

**Comment by nicolas-kuechler** [comment]
> Question here: Do we want to set the `docker_image_id` by default in new projects? i.e., add to `exp_base` in the cookie-cutter template?

I think this is the equivalent of the  `ec2_image_id` and should, therefore probably be defined in the same way:

https://github.com/nicolas-kuechler/doe-suite/blob/a5380076e0d4ae03493ae3ee2c13d6afac8c2434/cookiecutter-doe-suite-config/%7B%7Bcookiecutter.repo_name%20%7C%20default(%22doe-suite-config%22)%20%7D%7D/group_vars/%7B%7B%20cookiecutter.host_name%20%7D%7D/main.yml

But I think I would not have it as a cookie-cutter param because in 99% of cases people will just use the one you propose.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+3. Create a release from the tag on Github
+
+
+<!--
+## [Unreleased] - yyyy-mm-dd
+
+
+-->
+
+## [1.1.0] - yyyy-mm-dd
```
Comment (nicolas-kuechler) [inline CHANGELOG.md:18]: set in the end

File: demo_project/doe-suite-config/inventory/docker/Dockerfile-doe-ubuntu
[the remark this answers is not in the document]
Comment (nicolas-kuechler) [reply demo_project/doe-suite-config/inventory/docker/Dockerfile-doe-ubuntu:1]: @hiddely

File: Makefile [hunk 2]
```diff
@@ -242,7 +242,7 @@ clean-result-full: install
 clean-cloud: install
 	@cd $(does_config_dir) && \
 	ANSIBLE_CONFIG=$(PWD)/ansible.cfg \
-	poetry run ansible-playbook $(PWD)/src/clear.yml
+	poetry run ansible-playbook $(PWD)/src/clear.yml -e "cloud=$(cloud)"
 
 clean: clean-local-py clean-cloud
```
[the remark this answers is not in the document]
Comment (nicolas-kuechler) [reply Makefile:247]: @hiddely

File: docs/source/installation.rst [hunk 3]
```diff
@@ -40,6 +40,28 @@ ETHZ Euler - Specific Prerequisites
 To run experiments on ETHZ Euler, you must ensure that you can connect to ``euler.ethz.ch`` with SSH.
 Check the instructions provided by ETHZ Euler for `accessing the clusters using SSH <https://scicomp.ethz.ch/wiki/Accessing_the_clusters#SSH>`_.
 
+Docker - Specific Prerequisites
```
Comment (nicolas-kuechler) [inline docs/source/installation.rst:43]: @nicolas-kuechler still check

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicolas-kuechler) [reply Makefile]: -> done (`make clean` now asks the user whether it should clean docker and aws)

File: demo_project/doe-suite-config/inventory/docker/Dockerfile-doe-ubuntu
[the remark this answers is not in the document]
Comment (hiddely) [reply demo_project/doe-suite-config/inventory/docker/Dockerfile-doe-ubuntu:1]: TODO: Add comment to highlight custom dockerfiles can be included in this directory

File: src/roles/suite-cloud-instances-create/tasks/docker.yml [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+  when: "(does_config_dir + '/inventory/docker') is directory and (does_config_dir + '/inventory/docker/Dockerfile-' + item) is file"
+
+- name: Build docker images in resources/docker (if this fails, did you set DOES_DOCKER_SSH_PUBLIC_KEY?)
+  community.docker.docker_image:
+    build:
+      path: "{{ docker_build_dir }}"
+      dockerfile: "Dockerfile-{{ item }}"
+    name: "{{ item }}"
+    source: build
+    tag: "latest"
```
[the remark this answers is not in the document]
Comment (hiddely) [reply src/roles/suite-cloud-instances-create/tasks/docker.yml:92]: Yes

**Pull request merged**